src/coreclr/nativeaot/Runtime/threadstore.cpp

```diff
@@ -165,10 +165,21 @@ void ThreadStore::DetachCurrentThread()
     }
 
     {
+        // remove the thread from the list.
+        // this must be done even when shutdown is in progress.
+        // departing threads will be releasing their TLS storage and if not removed,
+        // will corrupt the thread list, while some prior to shutdown detach might still be in progress.
         ThreadStore* pTS = GetThreadStore();
         ReaderWriterLock::WriteHolder write(&pTS->m_Lock);
         ASSERT(rh::std::count(pTS->m_ThreadList.Begin(), pTS->m_ThreadList.End(), pDetachingThread) == 1);
         pTS->m_ThreadList.RemoveFirst(pDetachingThread);
+
+        // the rest of cleanup is not necessary if we are shutting down.
+        if (shutdownStarted)
+        {
+            return;
+        }
+
         pDetachingThread->Detach();
```

Comment:
We are going to skip `FixAllocContext` here. Is it going to cause problems/asserts with GC accounting if the GC manages to sneak in before the process actually exits?

Would it be better to only skip `Thread::Destroy`, but keep `Thread::Detach`?

Reply:
It would probably only affect statistics like total alloc bytes if someone manages to query for that before the exit. 
We may as well call `FixAllocContext`. It is not a lot of savings if we don't.